python-threatexchange/threatexchange/fetcher/apis/ncmec_api.py

```diff
@@ -0,0 +1,232 @@
+            ret[api.NCMECEntryType.image, "pdna"] = st
+            break
+    return ret
+
+
+NCMECUpdate = t.Dict[str, api.NCMECEntryUpdate]
+
+
+class NCMECSignalExchangeAPI(
+    fetch_api.SignalExchangeAPI[
+        NCMECCollabConfig,
+        NCMECCheckpoint,
+        NCMECSignalMetadata,
+        NCMECUpdate,
+    ]
+):
+    """
+    Conversion for the NCMEC hash API
+
+    Key implementation details:
+        1. API is a stream of content: opinion, hashes,
+           which need to be remapped to hash => opinion
+        2. Owners have ids
+        3. As of 5/2022 there are no false positive or seen statuses
+    """
+
+    def __init__(
+        self,
+        username: str = "",
+        password: str = "",
+    ) -> None:
+        super().__init__()
+        self._api = None
+        if username and password:
+            self._api = api.NCMECHashAPI(
+                username,
+                password,
+            )
+
+    @classmethod
+    def get_name(cls) -> str:
+        return _API_NAME
+
+    @classmethod
+    def get_config_class(cls) -> t.Type[NCMECCollabConfig]:
+        return NCMECCollabConfig
+
+    @classmethod
+    def get_checkpoint_cls(cls) -> t.Type[NCMECCheckpoint]:
+        return NCMECCheckpoint
+
+    @classmethod
+    def get_record_cls(cls) -> t.Type[NCMECSignalMetadata]:
+        return NCMECSignalMetadata
+
+    @property
+    def client(self) -> api.NCMECHashAPI:
+        if self._api is None:
+            raise Exception("NCMEC username and password not configured.")
+        return self._api
+
+    def fetch_iter(
+        self,
+        _supported_signal_types: t.Sequence[t.Type[SignalType]],
+        _collab: CollaborationConfigBase,
+        checkpoint: t.Optional[NCMECCheckpoint],
+    ) -> t.Iterator[
+        state.FetchDelta[t.Dict[str, api.NCMECEntryUpdate], NCMECCheckpoint]
+    ]:
+        start_time = 0
+        if checkpoint is not None:
+            start_time = checkpoint.max_timestamp
+        for result in self.client.get_entries_iter(start_timestamp=start_time):
+            yield state.FetchDelta(
+                {f"{entry.member_id}-{entry.id}": entry for entry in result.updates},
+                NCMECCheckpoint.from_ncmec_fetch(result),
+            )
+
+    @classmethod
+    def naive_fetch_merge(
+        cls,
+        old: t.Optional[NCMECUpdate],
+        new: NCMECUpdate,
+    ) -> NCMECUpdate:
+        old = old or {}
+        old.update(new)
+        return old
+
+    @classmethod
+    def naive_convert_to_signal_type(
+        cls,
+        signal_types: t.Sequence[t.Type[SignalType]],
+        fetched: NCMECUpdate,
+    ) -> t.Dict[t.Type[SignalType], t.Dict[str, NCMECSignalMetadata]]:
+        mapping: t.Mapping[
+            t.Tuple[api.NCMECEntryType, str], t.Type[SignalType]
+        ] = _get_conversion(signal_types)
+        ret: t.Dict[t.Type[SignalType], t.Dict[str, NCMECSignalMetadata]] = {}
+        for entry in fetched.values():
```

Comment:
How does NCMEC (and this API) handle deletes? I don't see any handling of 'entry == None'.

Reply:
search for `.deleted`